specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json

```diff
@@ -348,6 +348,251 @@
+            "type": "string",
+            "description": "Name of the managed HSM Pool"
+          },
+          {
+            "$ref": "#/parameters/MHSMPrivateEndpointConnectionName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "Private endpoint connection successfully returned.",
+            "schema": {
+              "$ref": "#/definitions/MHSMPrivateEndpointConnection"
+            }
+          },
+          "204": {
+            "description": "The private endpoint connection does not exist."
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ManagedHsmError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "MHSMPrivateEndpointConnections"
+        ],
+        "operationId": "MHSMPrivateEndpointConnections_Put",
+        "description": "Updates the specified private endpoint connection associated with the managed hsm pool.",
+        "x-ms-examples": {
+          "ManagedHsmPutPrivateEndpointConnection": {
+            "$ref": "./examples/ManagedHsm_putPrivateEndpointConnection.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ManagedHsmResourceGroupName"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "Name of the managed HSM Pool"
+          },
+          {
+            "$ref": "#/parameters/MHSMPrivateEndpointConnectionName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "properties",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/MHSMPrivateEndpointConnection"
+            },
+            "description": "The intended state of private endpoint connection."
+          }
+        ],
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "The state of private endpoint connection was updated successfully.",
+            "headers": {
+              "Retry-After": {
+                "description": "(specified only if operation does not finish synchronously) The recommended number of seconds to wait before calling the URI specified in Azure-AsyncOperation.",
+                "type": "integer",
+                "format": "int32"
+              },
+              "Azure-AsyncOperation": {
```

Comment:
If there is a change this finishes asynchronously, it should be authored as a long-running operation. The SDKs can't really deal with operations that are synchronous and asynchronous - only one or the other.

@JeffreyRichter @johanste is this normal in the management plane?

Reply:
We've run into a few services that want to return 200 or 202 but we've been steering them towards 202 always and never 200. That's what should happen here too.

Reply:
This is aligned with current AKV implementation which MHSM is also depending on. We can definitely improve this in future.